### ISSUE TITLE
[Q] Is there functionality to download artifacts only if they have not already been downloaded (especially for offline mode)?

### ISSUE THREAD

**Issue by kdu4108** [wandb/wandb#5309]
Hi! When using wandb artifacts, I run:
```
artifact = wandb.run.use_artifact("artifact1:latest")
artifact.download()
```
which downloads the artifact to `./artifacts/artifact1:v0`.

It appears that every time this script is rerun, it does some degree of redownloading (based on the printed text "wandb:   1 of 1 files downloaded. ").

Since I'm using artifacts to download datasets, it is a critical part of my program and I'd therefore like to be able to also run it in offline mode. Unfortunately, this results in the error "TypeError: Cannot use artifact when in offline mode." I understand that if the artifact has never been downloaded, then there's nothing we can do about that in offline mode and it should fail.

But, if the artifact has already been downloaded and we're in offline mode, would it instead be possible for wandb to return the path of the downloaded artifact? Are there any risks/reasons that discourage implementing this into the library?

Thanks!

**Comment by luisbergua** [comment]
Hi @kdu4108, thanks for your question! Would you mind explaining to me a little bit about your use case and what exactly are you trying to do so I can give you better advice? If I'm understanding you properly, you'd like to get the path if the artifact has been already downloaded when calling  `artifact.download()`, is it correct? This isn't possible because we don't store the path where the artifact is downloaded as it can be set with the argument `root` when calling [download](https://docs.wandb.ai/ref/python/artifact#download) so it can be different depending on this argument (by default it goes into a folder named `./artifacts`).

**Comment by kdu4108** [comment]
Yes thank you for the answer, that make sense! The download path is something the user controls, so it's their responsibility to know the path where the artifact should be downloaded/look in that location offline.

**Comment by moredatarequired** [comment]
@kdu4108 , I think this behavior should have been fixed in #5056 and made it into the 0.13.11 release, so if you're still having this issue please let me know!
